pallets/pooled-staking/src/calls.rs

```diff
@@ -0,0 +1,459 @@
+                (held, stake)
+            }
+            AllTargetPool::Leaving => {
+                let held = pools::Leaving::<T>::hold(&candidate, &delegator);
+                let shares = pools::Leaving::<T>::shares(&candidate, &delegator);
+                let stake = pools::Leaving::<T>::shares_to_stake(&candidate, shares)?;
+                pools::Leaving::<T>::set_hold(&candidate, &delegator, stake.clone());
+                (held, stake)
+            }
+        };
+
+        // Transfer is done using withdraw to ensure it works regardless of ED.
+        if let Some(diff) = stake.0.checked_sub(&held.0) {
+            T::Currency::transfer(
+                &T::StakingAccount::get(),
+                &delegator,
+                dbg!(diff),
+                Preservation::Preserve,
+            )?;
+            T::Currency::hold(&T::CurrencyHoldReason::get(), &delegator, diff)?;
+            return Ok(().into());
+        }
+
+        if let Some(diff) = held.0.checked_sub(&stake.0) {
+            T::Currency::release(
+                &T::CurrencyHoldReason::get(),
+                &delegator,
+                diff,
+                Precision::Exact,
+            )?;
+            T::Currency::transfer(
+                &delegator,
+                &T::StakingAccount::get(),
+                diff,
+                Preservation::Preserve,
+            )?;
+            return Ok(().into());
+        }
+
+        return Ok(().into());
+    }
+
+    pub fn request_delegate(
+        origin: OriginFor<T>,
+        candidate: Candidate<T>,
+        pool: TargetPool,
+        stake: T::Balance,
+    ) -> DispatchResultWithPostInfo {
+        let delegator = ensure_signed(origin)?;
+        ensure!(!Zero::is_zero(&stake), Error::<T>::StakeMustBeNonZero);
+
+        // Convert stake into joining shares quantity.
+        let shares = pools::Joining::<T>::stake_to_shares_or_init(&candidate, Stake(stake))?;
+
+        // If the amount was stake and is less than the value of 1 share it will round down to
+        // 0 share. We avoid doing any work for 0 shares.
+        ensure!(!Zero::is_zero(&shares.0), Error::<T>::StakeMustBeNonZero);
+
+        // We create the new joining shares. It returns the actual amount of stake those shares
+        // represents (due to rounding).
+        let stake = pools::Joining::<T>::add_shares(&candidate, &delegator, shares.clone())?;
+
+        // We hold the funds of the delegator and register its stake into the candidate stake.
+        T::Currency::hold(&T::CurrencyHoldReason::get(), &delegator, stake.0)?;
+        pools::Joining::<T>::increase_hold(&candidate, &delegator, &stake)?;
+        Candidates::<T>::add_total_stake(&candidate, &stake)?;
+
+        // We create/mutate a request for joining.
+        let now = frame_system::Pallet::<T>::block_number();
+        let operation_key = match pool {
+            TargetPool::AutoCompounding => PendingOperationKey::JoiningAutoCompounding {
```

Comment:
This means I can either select autoCompounding when I delegate or manual rewards, but not a percentage for each of them right? Just wondering how easy would be to add this in the future

Reply:
I guess one can always do 2 delegations targeting different pools?

Reply:
One would make 2 delegation requests, with the amounts being split according to the percentage you want.
We can add another extrinsic `request_delegate_split` that takes such percentage and make the 2 requests. We could also add a new request kind for that but that may introduce more complexity.

Reply:
we will do it if we find the need in the future, I was just wondering if it was possible. Right now the way it is its good, thanks!